core/dbt/parser/read_files.py

```diff
@@ -108,68 +117,44 @@ def get_source_files(project, paths, extension, parse_file_type, saved_files):
     return fb_list
 
 
-def read_files_for_parser(project, files, dirs, extension, parse_ft, saved_files):
-    parser_files = []
-    source_files = get_source_files(
-        project, dirs, extension, parse_ft, saved_files
-    )
-    for sf in source_files:
-        files[sf.file_id] = sf
-        parser_files.append(sf.file_id)
-    return parser_files
-
-
 # This needs to read files for multiple projects, so the 'files'
 # dictionary needs to be passed in. What determines the order of
 # the various projects? Is the root project always last? Do the
 # non-root projects need to be done separately in order?
-def read_files(project, files, parser_files, saved_files):
-
-    project_files = {}
-
-    project_files['MacroParser'] = read_files_for_parser(
-        project, files, project.macro_paths, '.sql', ParseFileType.Macro, saved_files
-    )
-
-    project_files['ModelParser'] = read_files_for_parser(
-        project, files, project.model_paths, '.sql', ParseFileType.Model, saved_files
-    )
-
-    project_files['SnapshotParser'] = read_files_for_parser(
-        project, files, project.snapshot_paths, '.sql', ParseFileType.Snapshot, saved_files
-    )
-
-    project_files['AnalysisParser'] = read_files_for_parser(
-        project, files, project.analysis_paths, '.sql', ParseFileType.Analysis, saved_files
-    )
-
-    project_files['SingularTestParser'] = read_files_for_parser(
-        project, files, project.test_paths, '.sql', ParseFileType.SingularTest, saved_files
-    )
-
-    # all generic tests within /tests must be nested under a /generic subfolder
-    project_files['GenericTestParser'] = read_files_for_parser(
-        project, files, project.generic_test_paths, '.sql', ParseFileType.GenericTest, saved_files
-    )
-
-    project_files['SeedParser'] = read_files_for_parser(
-        project, files, project.seed_paths, '.csv', ParseFileType.Seed, saved_files
-    )
-
-    project_files['DocumentationParser'] = read_files_for_parser(
-        project, files, project.docs_paths, '.md', ParseFileType.Documentation, saved_files
-    )
-
-    project_files['SchemaParser'] = read_files_for_parser(
-        project, files, project.all_source_paths, '.yml', ParseFileType.Schema, saved_files
-    )
-
-    # Also read .yaml files for schema files. Might be better to change
-    # 'read_files_for_parser' accept an array in the future.
-    yaml_files = read_files_for_parser(
-        project, files, project.all_source_paths, '.yaml', ParseFileType.Schema, saved_files
-    )
-    project_files['SchemaParser'].extend(yaml_files)
-
-    # Store the parser files for this particular project
-    parser_files[project.project_name] = project_files
+def read_files(
```

Comment:
Not that this is a **breaking** API change of this function. What is dbt's policy regarding such functions, as the public API of dbt is the CLI, but this function might still be used by some users that call `dbt-core` programatically?